sql/core/src/main/scala/org/apache/spark/sql/util/QueryExecutionListener.scala

```diff
@@ -109,6 +109,16 @@ class ExecutionListenerManager private[sql](session: SparkSession, loadExtension
     listenerBus.removeAllListeners()
   }
 
+  /**
+   * Removes [[ExecutionListenerBus]] from LiveListenerBus.queues. people should call this function
+   * when the manager belongs session was removed.
+   */
+  @DeveloperApi
+  def clearListenerBus(): Unit = {
+    session.sparkContext.removeSparkListener(listenerBus)
```

Comment:
The problem here is that it removes all the listeners even added from another session.

Reply:
+1

Reply:
It only removes the session belongs `QueryExecutionListener`, not the others. I can't imagine how it can remove listener create by other sessions, can you give me an example?

Reply:
You're removing all listeners from Spark Context which other Spark sessions share.

Reply:
Add a new test and explain how we don't remove the Spark Context shared listener.

Reply:
Oh, okay. I misread.